conda/models/records.py

```diff
@@ -281,7 +281,7 @@ def __eq__(self, other):
         return self._pkey == other._pkey
 
     def dist_str(self):
-        return "%s::%s-%s-%s" % (self.channel.canonical_name, self.name, self.version, self.build)
+        return "%s/%s::%s-%s-%s" % (self.channel.canonical_name, self.subdir, self.name, self.version, self.build)
```

Comment:
A question: is it worthwhile to be more defensive here if `self.subdir` is `None`?

`SubDir` is not a nullable field, but this line seems to imply it's possible to return None, as `Channel`'s default platform is None:
https://github.com/conda/conda/blob/master/conda/models/records.py#L148

Reply:
It does seem wise.  I'm not sure what the appropriate behavior is, though.  If you don't know it at all, it could be either the native platform or noarch.  What's a better guess?  It used to be that the native platform was almost always going to be right, but more packages are switching to noarch now to ease the load on build servers.

Reply:
Hm, what about falling back to the old behavior (of not including the subdir in the `dist_str`)? That way it's not strictly worse for a user that encounters a package with no subdir.

Reply:
Seems fine to me.  Maybe something like this:
```suggestion
        return "%s%s::%s-%s-%s" % (self.channel.canonical_name, ("/" + self.subdir) if self.subdir else "", self.name, self.version, self.build)
```